recipes-extended/images/pelux-image-update/intel-corei7-64/emmcsetup2.lua

```diff
@@ -19,109 +14,113 @@ function cmdexec(cmd)
 
-	local eMMC = "/dev/sda"
-	ret = file_exists("/dev/sda")
+	dev = os.capture("lsblk -no PARTLABEL,KNAME | grep " .. label .. " | awk '{print $2}'")
 
-	if (ret == false) then
-		return false, "Cannot find eMMC"
-	end
+	return pdev, dev
+end
 
-	ret, out = cmdexec("/usr/sbin/sfdisk -d " .. eMMC .. "> /tmp/dumppartitions")
-	if (false == ret) then
-		return ret, out
+function get_update_device_from_boot_device(dev)
+	local part = ""
+	local length = dev:len()-1
+	local num = dev:sub(length, length)
+
+	if ("2" == num) then
+		part = "3"
+	elseif ("3" == num) then
+		part = "2"
+	else
+		return false, "Incorrect Partition number " .. num
 	end
 
-	-- check if there are two identical partitions
-	-- and create the second one if no available
-	f = io.input("/tmp/dumppartitions")
-	fo = io.output("/tmp/partitions")
-	t = f:read()
-	found = false
-	while (t ~= nil) do
-		j=0
-		j=string.find(t, "/dev/sda3")
-		ret, out = fo:write(t .. "\n")
-		if (ret == nil) then
-			fo:close()
-			f:close()
-			return false, out
-		end
-		if (j == 1) then
-			found=true
-			break
-		end
-		j=string.find(t, "/dev/sda2")
-		if (j == 1) then
-			start, size = string.match(t, "%a+%s*=%s*(%d+), size=%s*(%d+)")
-		end
-		t = f:read()
-	end
+	local updatedev = dev:sub(1, length-1) .. part
 
-	if (found) then
-		f:close()
-		fo:close()
-		return true, out
-	end
+	return true, updatedev
+end
 
-	start=start+size
-	partitions = eMMC .. "3 : start=    " .. string.format("%d", start) .. ", size=  " .. size .. ", type=83\n"
+function get_update_device_from_boot_device_label(label)
+	local ret
+	local pdev
+	local bootdev
+	local updatedev
 
-	ret, out = fo:write(partitions)
-	fo:close()
-	f:close()
+	pdev, bootdev = get_device_from_boot_device_label(label)
 
-	if (ret == nil) then
-		return false, out
-	end
+	ret, updatedev = get_update_device_from_boot_device(bootdev)
+
+	print("Boot Device:", bootdev, "Update Device:", updatedev)
 
-	out = os.capture("/usr/sbin/sfdisk --force " .. eMMC .. " < /tmp/partitions")
+	return ret, updatedev
+end
+
+function preinst()
+	local out = "Pre installed script called"
+	local ret
+	local updatedev
 
-	-- use partprobe to inform the kernel of the new partitions
+	ret, updatedev = get_update_device_from_boot_device_label("platform1")
+	if (ret == false) then
+		return ret, updatedev .. ", Cannot find update device"
+	end
 
-	ret, out = cmdexec("/usr/sbin/partprobe " .. eMMC)
+	ret, out = cmdexec("ln -s /dev/" .. updatedev .. " /dev/updatedev")
 	if (false == ret) then
 		return ret, out
 	end
 
-	return true, out
+	return ret, out
 end
 
 function postinst()
 	local out = "Post installed script called"
+	local ret
+	local updatedev
+
+	ret, updatedev = get_update_device_from_boot_device_label("platform1")
+	if (ret == false) then
+		return ret, updatedev .. ", Cannot find update device"
+	end
+
+	local mountpoint = "/tmp/mounted" .. updatedev
+	local mountdev = "/dev/" .. updatedev
```

Comment:
Perhaps explain that /dev/updatedev is needed due to it not being possible to use labels or UUID:s in sw-description?